recipe/meta.yaml

```
@@ -12,35 +12,42 @@ source:
   sha256: e8e1017b2cf1dda767aef19d2fd9ee5ebe07e050d430f77a0a7c66ba08b8cdae
 
 build:
-  number: 1
-  noarch: python
+  number: 0
   script:
-    # We are currently building a noarch package, but pysoundfile contains a
-    # install-time check for the OS and injects the sf_wchar_open function
-    # conditionally. If this variable is not set, opening files on Windows doesn't work
-    # Fix coming hopefully with https://github.com/bastibe/python-soundfile/pull/288
-    # Original issue on conda-forge: https://github.com/conda-forge/libsndfile-feedstock/issues/14
-    - export PYSOUNDFILE_PLATFORM=win32
-    - python -m pip install . --no-deps -vv
+      # To make sure we really do not bring in _soundfile_data
+    - rm -rf _soundfile_data
+    - python -m pip install . -vv --no-deps --no-build-isolation
+  skip: true  # [py<36 or s390x or win]
+  patches:
+    - 0001-unvendoring-libsndfile.patch
 
 requirements:
+  build:
+    - patch     # [not win]
+    - m2-patch  # [win]
   host:
     - pip
-    - python >=3.6
+    - python
     - setuptools
+    - wheel
     - cffi
   run:
-    - python >=3.6
+    - python
     - cffi
     - numpy
     - libsndfile >=1.2
 
 test:
+  source_files:
+    - tests
+  requires:
+    - pytest
   imports:
     - soundfile
   commands:
-    # Verify that soundfile is linking the correct libsndfile object
-     python -c "import soundfile; print(soundfile._libname)"
+     # Verify that soundfile is linking the correct libsndfile object
+     - python -c "import soundfile; print(soundfile._libname)"
```

what should be the correct libname? It might be nicer to fail if not correct

---

This came in when forked from conda-forge. I can remove it and see if it's necessary.

---

You could pipe the output from this into something that checks that the text output from `print(soundfile._libname)` is correct

---

for linux is something like `$PREFIX/lib/libsndfile.so` and for `osx` it prints `$PREFIX/bin/../lib/libsndfile.dylib`, windows is skipped otherwise it would have a `.dll` file.

you could go for something like:

```suggestion
     - python -c "import soundfile; from pathlib import Path; print(soundfile._libname); assert(Path(soundfile._libname).stem == 'libsndfile')"
```